dash/development/_r_components_generation.py

```diff
@@ -0,0 +1,531 @@
+        prefix
+    )
+    file_name = "{}{}.R".format(prefix, name)
+
+    file_path = os.path.join('R', file_name)
+    with open(file_path, 'w') as f:
+        f.write(import_string)
+        f.write(class_string)
+
+    # generate the R help pages for each of the Dash components that we
+    # are transpiling -- this is done to avoid using Roxygen2 syntax,
+    # we may eventually be able to generate similar documentation using
+    # doxygen and an R plugin, but for now we'll just do it on our own
+    # from within Python
+    write_help_file(
+        name,
+        props,
+        description,
+        prefix
+    )
+
+    print('Generated {}'.format(file_name))
+
+
+# pylint: disable=inconsistent-return-statements
+def generate_export_string(name, prefix):
+    if not name.endswith('-*') and \
+            str(name) not in r_keywords and \
+            str(name) not in ['setProps', 'children', 'dashEvents']:
+        return 'export({}{})\n'.format(prefix, name)
+
+
+def write_js_metadata(project_shortname):
+    """
+    Write an internal (not exported) R function to return all JS
+    dependencies as required by htmltools package given a
+    function string
+
+    Parameters
+    ----------
+    project_shortname = hyphenated string, e.g. dash-html-components
+
+    Returns
+    -------
+
+    """
+    function_string = generate_js_metadata(
+        project_shortname
+    )
+    file_name = "internal.R"
+
+    # the R source directory for the package won't exist on first call
+    # create the R directory if it is missing
+    if not os.path.exists('R'):
+        os.makedirs('R')
+
+    file_path = os.path.join('R', file_name)
+    with open(file_path, 'w') as f:
+        f.write(function_string)
+
+    # now copy over all JS dependencies from the (Python) components dir
+    # the inst/lib directory for the package won't exist on first call
+    # create this directory if it is missing
+    if not os.path.exists('inst'):
+        os.makedirs('inst')
+
+    if not os.path.exists('inst/lib'):
+        os.makedirs('inst/lib')
+
+    for javascript in glob.glob('{}/*.js'.format(project_shortname)):
+        shutil.copy(javascript, 'inst/lib/')
+
+    for css in glob.glob('{}/*.css'.format(project_shortname)):
+        shutil.copy(css, 'inst/lib/')
+
+
+# pylint: disable=R0914
+def generate_rpkg(pkg_data,
+                  project_shortname,
+                  export_string):
+    """
+    Generate documents for R package creation
+
+    Parameters
+    ----------
+    pkg_data
+    project_shortname
+    export_string
+
+    Returns
+    -------
+
+    """
+    # Leverage package.json to import specifics which are also applicable
+    # to R package that we're generating here, use .get in case the key
+    # does not exist in package.json
+
+    package_name = snake_case_to_camel_case(project_shortname)
+    package_description = pkg_data.get('description', '')
+    package_version = pkg_data.get('version', '0.01')
```

Comment:
Default should be `0.0.1` and it's a required field in `package.json` so it's unlikely that it's not there.

Reply:
OK, I've changed the version number to `0.0.1`, even if unlikely it's not there, will still set the default.